code/controllers/subsystems/serialization.dm

```diff
@@ -0,0 +1,316 @@
+
+/obj/item/stack/material/finalize_deserialized_data(list/_options)
+    if(ispath(material)) // validate
+      set_material(material)
+
+*/
+
+// Serializes a specific object to a string.
+// Use `write_json(json_string, file_path)` to save the string produced by this to disk.
+/datum/controller/subsystem/serialization/proc/object_to_json(datum/object, make_pretty = TRUE, list/options)
+	if(isnull(object))
+		return
+	var/list/data = object.save_serialized_data(options)
+	var/json_string = json_encode(data)
+	if(make_pretty)
+		json_string = pretty_json(json_string)
+	return json_string
+
+// Deserializes a string and attempts to recreate the object that had made it.
+// Returns the resulting object if successful.
+// Use `read_json(file_path)` to get the string that this proc needs.
+/datum/controller/subsystem/serialization/proc/json_to_object(json_string, list/options)
+	var/list/data = json_decode(json_string)
+	if(!istype(data))
+		log_debug("DESERIALIZATION: Supplied json string resulted in no data.")
+		return
+	return deserialize_list(data, options)
+
+
+
+/datum/controller/subsystem/serialization/proc/deserialize_list(list/data, list/options)
+	var/object_type = data[NAMEOF(src, type)]
+	if(!ispath(text2path(object_type)))
+		log_debug("DESERIALIZATION: Object type '[object_type]' does not appear to exist.")
+		// Consider adding a method to migrate old paths to new ones in some kind of list?
+		return
+	
+	var/datum/object = new object_type()
+	var/address = data[NAMEOF(src, persistent_address)]
+//	object.load_deserialized_data(data, options)
+	object.iterate_deserialized_variables(data, options)
+	deserialized_objects[address] = weakref(object)
+	most_recent_deserialized_data = data.Copy()
+//	process_reference_lookup_queue()
+	return object
+
+// File I/O
+
+// Writes a json string to disk, storing it permanently.
+// This proc will happily overwrite whatever is in the path, so be aware of that.
+/datum/controller/subsystem/serialization/proc/write_json(json_string, file_path)
+	var/json_file = file(file_path)
+	if(fexists(json_file))
+		fdel(json_file) // This stops the file from getting appended, and breaking the json parser.
+	//to_file(json_file, json_string)
+	text2file(json_string, json_file)
+
+// Reads a json file from disk, retriving the json string contained inside.
+/datum/controller/subsystem/serialization/proc/read_json(file_path)
+	if(!fexists(file_path))
+		CRASH("Tried to open file '[file_path]', but it does not exist or cannot be accessed.")
+	
+	var/json_file = file(file_path)
+	return file2text(json_file)
+
+// Adds some whitespace to make the json formatting easier for humans to read.
+// `json_decode()` ignores it so it's purely for the benefit of humans.
+// However it will add a small amount of overhead when something is being saved.
+// The file size will also be slightly bigger due to the added whitespace.
+/datum/controller/subsystem/serialization/proc/pretty_json(json_string)
```

Comment:
Might be worth gating this behind an #ifdef or config or something, this is significantly more than `a little` overhead due to all the string operations.